File: ios/Classes/NotificationTapHandler.swift
import Foundation
import Flutter

class NotificationTapHandler: NSObject, FlutterStreamHandler {
    static var eventSink: FlutterEventSink?
    static var bufferedPayload: [String: Any]? = nil

    static func register(with registrar: FlutterPluginRegistrar) {
        let eventChannel = FlutterEventChannel(name: "pnta_flutter/notification_tap", binaryMessenger: registrar.messenger())
        let instance = NotificationTapHandler()
        eventChannel.setStreamHandler(instance)
    }

    static func sendTapPayload(_ payload: [String: Any]) {
        if let sink = eventSink {
            sink(payload)
        } else {
            bufferedPayload = payload
        }
    }

    // FlutterStreamHandler
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        NotificationTapHandler.eventSink = events
        if let payload = NotificationTapHandler.bufferedPayload {
            events(payload)
            NotificationTapHandler.bufferedPayload = nil
        }
        return nil
    }

    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        NotificationTapHandler.eventSink = nil
        return nil
    }
} 

File: android/src/main/kotlin/io/pnta/pnta_flutter/NotificationTapHandler.kt
package io.pnta.pnta_flutter

import io.flutter.plugin.common.EventChannel
import io.flutter.plugin.common.BinaryMessenger

object NotificationTapHandler : EventChannel.StreamHandler {
    private var eventSink: EventChannel.EventSink? = null
    private var bufferedPayload: Map<String, Any>? = null

    fun register(messenger: BinaryMessenger) {
        val eventChannel = EventChannel(messenger, "pnta_flutter/notification_tap")
        eventChannel.setStreamHandler(this)
    }

    fun sendTapPayload(payload: Map<String, Any>) {
        if (eventSink != null) {
            eventSink?.success(payload)
        } else {
            bufferedPayload = payload
        }
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
        bufferedPayload?.let {
            events?.success(it)
            bufferedPayload = null
        }
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }
} 